libs/common/Generator.cs

```diff
@@ -29,10 +40,9 @@ public static string CreateHexId(int size = 40)
         /// <param name="size">The length of the hex identifier string</param>
         /// <returns></returns>
         public static string DefaultHexId(int size = 40)
-        {
-            Span<byte> nodeIdBuffer = stackalloc byte[size / 2];
-            nodeIdBuffer.Clear();
-            return Convert.ToHexString(nodeIdBuffer).ToLowerInvariant();
-        }
+            => size == 40 ? DefaultHexId40 : new string('0', size & ~1); // note: trimmed to multiples of 2
+
+        const string DefaultHexId40 = "0000000000000000000000000000000000000000"; // '0' x40
```

Comment:
Change x40 to something like 40 times?
I literally read the comment as 0x40 and counted the zeros and asked myself it's not 64 chars.